Engine/Module/ModuleResourceManager.cpp

```diff
@@ -39,136 +79,244 @@ update_status ModuleResourceManager::PreUpdate()
-	 {
-		 Sleep(1000);
-	 } 
-	 thread_comunication.main_importing_hash = std::hash<std::string>{}(file.file_path);
-	 ImportResult  result = InternalImport(file, force);
-	 thread_comunication.main_importing_hash = 0;
-	 return result;
- }
-
+void ModuleResourceManager::CleanInconsistenciesInDirectory(const Path& directory_path)
+{
+	for (auto & path_child : directory_path.children)
+	{
+		if (thread_comunication.stop_thread)
+		{
+			return;
+		}
+		thread_comunication.thread_importing_hash = std::hash<std::string>{}(path_child->GetFullPath());
+		while (thread_comunication.main_importing_hash == std::hash<std::string>{}(directory_path.GetFullPath()))
+		{
+			Sleep(1000);
+		}
 
- void ModuleResourceManager::CreatePrefab(const std::string &path, GameObject * gameobject_to_save) const
- {
-	prefab_importer->CreatePrefabResource(File(path),gameobject_to_save);
- }
+		if (path_child->IsDirectory())
+		{
+			CleanInconsistenciesInDirectory(*path_child);
+		}
+		else if (path_child->IsMeta())
+		{
+			if (!metafile_manager->IsMetafileConsistent(*path_child))
+			{
+				metafile_manager->DeleteMetafileInconsistencies(*path_child); // TODO: This causes memory violations because bector size is modified. Look for a better way of deleting files.
+			}
+		}
+		thread_comunication.thread_importing_hash = 0;
+	}
+}
 
-void ModuleResourceManager::ImportAllFilesInDirectory(const File& file, bool force)
+void ModuleResourceManager::ImportAssetsInDirectory(const Path& directory_path)
  {
-	 for (auto & child : file.children)
+	 for (size_t i = 0; i < directory_path.children.size(); ++i)
 	 {
+		 Path* path_child = directory_path.children[i];
 		 if (thread_comunication.stop_thread)
 		 {
 			 return;
 		 }
-		 thread_comunication.thread_importing_hash = std::hash<std::string>{}(child->file_path);
-		 while (thread_comunication.main_importing_hash == std::hash<std::string>{}(file.file_path))
+		 /*
+		 thread_comunication.thread_importing_hash = std::hash<std::string>{}(path_child->GetFullPath());
+		 while (thread_comunication.main_importing_hash == std::hash<std::string>{}(directory_path.GetFullPath()))
 		 {
 			 Sleep(1000);
 		 }
-		 bool is_directory = child->file_type == FileType::DIRECTORY;
-		 if (is_directory && !default_importer->Import(*child.get(), force).succes)
+		 */
+		 if (path_child->IsDirectory())
 		 {
-			 ImportAllFilesInDirectory(*child.get(), force);
+			 ImportAssetsInDirectory(*path_child);
 		 }
-		 else if (!is_directory)
+		 else if (path_child->IsImportable())
 		 {
-			 InternalImport(*child.get(), force);
+			 Import(*path_child);
+			 ++thread_comunication.loaded_items;
 		 }
-		 ++thread_comunication.loaded_items;
 		 thread_comunication.thread_importing_hash = 0;
 	 }
  }
 
-
-ImportResult ModuleResourceManager::InternalImport(const File& file, bool force) const
+uint32_t ModuleResourceManager::Import(Path& file_path)
 {
-	ImportResult result;
-	std::lock_guard<std::mutex> lock(thread_comunication.thread_mutex);
```

Comment:
It is important to have it.

Reply:
I wanted to comment with you this, lets talk it by discord